js/Header/SearchBar/SearchLayer.js

```diff
@@ -0,0 +1,69 @@
+export class SearchLayer {
+  constructor($target) {
+    this.$target = $target;
+  }
+
+  async render(data) {
+    const renderData = await data;
+    this.$target.innerHTML = this.template(renderData);
+  }
+
+  template(searchData) {
+    const { searchHistory, suggestion } = searchData;
+    return `<ul class="search-bar__result-container">
+              ${this.makeRecentHistoryTemplate(searchHistory)}
+              ${this.makeTrendingSuggestionTemplate(suggestion)}
+            </ul>`;
+  }
+
+  makeRecentHistoryTemplate(data) {
+    const historyData = data.toReversed().slice(0, 5);
+    return `${historyData
+      .map(
+        (item) =>
+          `<li class="search-bar__result recentSearch">
+            <a href="#">${item.content}</a>
+            <button>
+              <img src="./assets/icons/close.svg" alt="" />
+            </button>  
+          </li>`,
+      )
+      .join('')}`;
+  }
+
+  makeTrendingSuggestionTemplate(data) {
+    return `${data
+      .map(
+        (item) =>
+          `<li class="search-bar__result suggestion">
+            <img src="./assets/icons/arrow-top-right.svg" alt="" />
+            <a href="#">${item.content}</a>
+          </li>`,
+      )
+      .join('')}`;
+  }
+
+  // makeAutoSuggestionTemplate() {
+  // 아직 구현중...
+  // }
+
+  updateLayer(model) {
+    const { searchData } = model;
+    this.render(searchData);
+  }
```

Comment:
화면을 리렌더링하는 데에 필요한 것은 데이터 이지요?
구현해주신 SearchModel은 데이터 말고도 다양한 메서드를 가지고 있구요!
그래서 model 전체를 넣어주게 되면 SearchLayer에서 SearchModel에게 불필요한 의존성을 가지게 된다고 생각합니다.

모델 전체를 넘겨주기보다는 진짜 필요한 정보만 넘겨주는 건 어떨까요?
```suggestion
  updateLayer(searchData) {
    this.render(searchData);
  }
```

Reply:
말씀해주신대로 모든 데이터가 아닌 필요한 데이터만 넘겨주는게 불필요한 의존성을 줄일 수 있을 것 같네요. 수정하겠습니다!